CHANGELOG.rst

```diff
@@ -1,6 +1,24 @@
 Changelog
 =========
 
+1.4.1rc1 (2024-05-16)
```

Comment:
Do we need to preserve Changelogs for RCs? 

Also adding this comment to update & confirm the release date if we plan to update this 1.4.1 instead of 1.4.1rc and also based on when we merge/release.

Reply:
I like the idea, totally makes sense to me.